api/jobs/handlers.py

```diff
@@ -280,6 +287,9 @@ def add(self):
         if submit.get('destination', None) is not None:
             destination = create_containerreference_from_dictionary(submit['destination'])
         else:
+            if len(inputs.keys()) < 1:
+                raise Exception('No destination specified and no inputs to derive from')
```

Comment:
We shouldn't 500 because user data wasn't formatted as expected. 400 or something like an `InputValidationException` would be more helpful.